laszip fails to build on linux.

```
/build/source/src/lasunzipper.cpp:79:46: error: ‘numeric_limits’ is not a member of ‘std’
     79 |     size_t max_bytes_in_output_stream = std::numeric_limits<std::stringstream::int_type>::max();
```

Missing an include? Not choosing a modern enough c++ std?

--- a/pkgs/development/python-modules/laspy/default.nix
+++ b/pkgs/development/python-modules/laspy/default.nix
@@ -3,6 +3,7 @@
 , fetchPypi
 , numpy
 , laszip
+, setuptools
 , pytestCheckHook
 , pythonOlder
 }:
@@ -10,7 +11,7 @@
 buildPythonPackage rec {
   pname = "laspy";
   version = "2.4.1";
-  format = "setuptools";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
 
@@ -19,12 +20,16 @@ buildPythonPackage rec {
     hash = "sha256-E8rsxzJcsiQsslOUmE0hs7X3lsiLy0S8LtLTzxuXKsQ=";
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+
   propagatedBuildInputs = [
     numpy
     laszip
   ];
 
-  checkInputs = [
+  nativeCheckInputs = [
     pytestCheckHook
   ];
 
@@ -36,7 +41,7 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "Interface for reading/modifying/creating .LAS LIDAR files";
     homepage = "https://github.com/laspy/laspy";
-    changelog = "https://github.com/laspy/laspy/blob/2.4.1/CHANGELOG.md";
+    changelog = "https://github.com/laspy/laspy/blob/${version}/CHANGELOG.md";
     license = licenses.bsd2;
     maintainers = with maintainers; [ matthewcroughan ];
   };

--- a/pkgs/development/python-modules/laspy/default.nix
+++ b/pkgs/development/python-modules/laspy/default.nix
@@ -10,14 +10,14 @@
 
 buildPythonPackage rec {
   pname = "laspy";
-  version = "2.4.1";
+  version = "2.5.1";
   format = "pyproject";
 
   disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-E8rsxzJcsiQsslOUmE0hs7X3lsiLy0S8LtLTzxuXKsQ=";
+    hash = "sha256-uqPJxswVVjbxYRSREfnPwkPb0U9synKclLNWsxxmjy4=";
   };
 
   nativeBuildInputs = [

--- a/pkgs/development/python-modules/laszip/default.nix
+++ b/pkgs/development/python-modules/laszip/default.nix
@@ -1,4 +1,5 @@
 { lib
+, stdenv
 , buildPythonPackage
 , fetchFromGitHub
 , scikit-build-core
@@ -10,7 +11,7 @@
 
 buildPythonPackage rec {
   pname = "laszip-python";
-  version = "0.2.1";
+  version = "0.2.3";
 
   format = "pyproject";
 
@@ -20,8 +21,10 @@ buildPythonPackage rec {
     owner = "tmontaigu";
     repo = pname;
     rev = version;
-    hash = "sha256-ujKoUm2Btu25T7ZrSGqjRc3NR1qqsQU8OwHQDSx8grY=";
+    hash = "sha256-MiPzL9TDCf1xnCv7apwdfcpkFnBRi4PO/atTQxqL8cw=";
   };
+
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-std=c++17";
 
   nativeBuildInputs = [
     cmake

